frontend/src/components/Searchbar/Searchbar.tsx

```diff
@@ -58,51 +73,113 @@ export interface SearchbarProps extends Omit<InputProps, 'onChange'> {
   value?: string
 
   /**
-   * Function to be invoked when the value in the searchbar input changes (but
-   * the search button has not been clicked).
+   * Optional. Function to be invoked when the value in the searchbar input
+   * changes (but the search button has not been clicked). Note: at least one of
+   * onSearch and onChange must be provided for the searchbar to work properly!
+   * @param newValue value of the search input
    */
   onChange?: (newValue: string) => void
+
+  /**
+   * Optional. If provided, adds a filter button to the searchbar. The initial
+   * (default) value of the filter, which is most likely the option for "No filter".
+   */
+  filterValue?: string
+
+  /**
+   * Optional. The remainder of the filter options, not including the default
+   * filterValue. Will be ignored unless filterValue has been provided.
+   */
+  filterOptions?: string[]
+
+  /**
+   * Optional. Function to be invoked when a filter option has been selected.
+   * Will be ignored unless filterValue has been provided.
```

Comment:
I'm wondering if we can design a more intuitive API for indicating "No filter". intuitively the behaviour I'd expect from this component would be something like:
- as long as I provide `filterOptions` and `onFilter`, the filter button gets added
- if I don't provide `filterValue`, the component defaults to "no filter"

wdyt?

Reply:
implemented this for now in [72a9d92](https://github.com/opengovsg/FormSG/pull/5458/commits/72a9d92bca2923427e1c5f8dbf187bf4b5bdb538), feel free to give more comments!

Reply:
yup sgtm, and the docs can clarify that `filterValue` is ignored if it is not an element of `filterOptions`